Wrappers/Python/cil/optimisation/operators/GradientOperator.py

```diff
@@ -74,15 +74,23 @@ def __init__(self, domain_geometry, method = 'forward', bnd_cond = 'Neumann', **
         """Constructor method
         """        
         
+        # Default backend = C
         backend = kwargs.get('backend',C)
 
+        # Default correlation for the gradient coupling
         correlation = kwargs.get('correlation',CORRELATION_SPACE)
-
+        
+        # Add attributes for pseudo CIL geometries, e.g., SIRF data
+        if not isinstance(domain_geometry, ImageGeometry):
+            domain_geometry.channels = 1
+            domain_geometry.dimension_labels = [None]*len(domain_geometry.shape)
+        
+        # Space correlation on multichannel data call numpy backend
         if correlation == CORRELATION_SPACE and domain_geometry.channels > 1:
             #numpy implementation only for now
             backend = NUMPY
             warnings.warn("Warning: correlation='Space' on multi-channel dataset will use `numpy` backend")
-           
+        
         if method != 'forward':
             backend = NUMPY
             warnings.warn("Warning: method = {} implemented on `numpy` backend. Other methods are backward/centered.".format(method))            
```

Comment:
Somewhere here we need to act if the data type is complex, and set `backend = NUMPY`

Reply:
Complex case will be treated in another PR. Currently there is one option that will work for both backends but will require some work from the user.

```python

    # option A
    ig1 = ImageGeometry(3,3)
    ig2 = ImageGeometry(3,3)
    bg = BlockGeometry(ig1,ig2)
    Grad1 = GradientOperator(ig1, backend='c') # or numpy
    Grad2 = GradientOperator(ig2, backend='c') # or numpy
    Zero1 = ZeroOperator(ig1, Grad1.range)
    BlockOp = BlockOperator(Grad1, Zero1, Zero1, Grad2, shape=(2,2))
    x_np = np.random.random(ig1.shape) + np.random.random(ig2.shape) * 1j
    x_cil = bg.allocate()
    x_cil[0].fill(np.real(x_np))
    x_cil[1].fill(np.imag(x_np))
    res1 = BlockOp.direct(x_cil)

```